### PR TITLE
Week 9 Assignment To Do App

### PR DESCRIPTION
**Pull request by jimishgajjar**
Week 9 To Do Application Assignment Using React Class Component, States & Props.

### CODE REVIEW

File: Week-9/to-do-app/src/components/containers/TodoContainer.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+      if (itemId === item.id) item.done = !item.done;
+
+      return item;
+    });
+
+    this.alertAudio.play();
+
+    // State Updates are Merged
+    this.setState({
+      todos: [].concat(updatedItems),
```
Comment (subhadip-pesto) [inline Week-9/to-do-app/src/components/containers/TodoContainer.js:95]: this.setState({todos: updatedItems});

File: Week-9/to-do-app/src/components/containers/TodoContainer.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    });
+  };
+
+  taskDelete = (itemId) => {
+    var updatedItems = this.state.todos.filter((item) => {
+      return item.id !== itemId;
+    });
+
+    this.setState({
+      todos: [].concat(updatedItems),
```
Comment (subhadip-pesto) [inline Week-9/to-do-app/src/components/containers/TodoContainer.js:116]: same as above

File: Week-9/to-do-app/src/components/containers/TodoContainer.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+    );
+    if (todoItemIndex !== -1) {
+      const todo = this.state.todos[todoItemIndex];
+      todo.label = this.state.labelVal;
+      todo.description = this.state.discriptionVal;
+
+      this.setState({
+        labelVal: "",
+        discriptionVal: "",
+        editId: "",
```
Comment (subhadip-pesto) [inline Week-9/to-do-app/src/components/containers/TodoContainer.js:166]: add this.setState({
  labelVal: "",
  discriptionVal: "",
  editId: "",
 todos: this.state.todos
})

File: Week-9/to-do-app/src/components/containers/TodoContainer.js [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+        autoClose: 2000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+        theme: "dark",
+      });
+
+      return todo;
```
Comment (subhadip-pesto) [inline Week-9/to-do-app/src/components/containers/TodoContainer.js:180]: why you are returning this?

File: Week-9/to-do-app/src/components/containers/TodoContainer.js [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+    }
+  };
+
+  TaskUndo = (itemId) => {
```
Comment (subhadip-pesto) [inline Week-9/to-do-app/src/components/containers/TodoContainer.js:184]: change function name

File: Week-9/to-do-app/src/components/containers/TodoContainer.js [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+    var undoItems = this.state.todos.map((item) => {
+      if (itemId === item.id) item.done = !item.done;
+
+      return item;
+    });
+
+    // State Updates are Merged
+    this.setState({
+      todos: [].concat(undoItems),
```
Comment (subhadip-pesto) [inline Week-9/to-do-app/src/components/containers/TodoContainer.js:193]: same as above mentioned comments

File: Week-9/to-do-app/src/components/containers/TodoContainer.js (hunk 1 above)
Comment (subhadip-pesto) [inline Week-9/to-do-app/src/components/containers/TodoContainer.js:95]: Fixx all given comments

**Review by subhadip-pesto — CHANGES_REQUESTED** [review]

File: Week-9/to-do-app/src/components/containers/TodoContainer.js (hunk 1 above)
Comment (jimishgajjar) [reply Week-9/to-do-app/src/components/containers/TodoContainer.js]: Changes Updated

**Review by subhadip-pesto — APPROVED** [review]

**Pull request merged**